selfservice/flow/settings/handler_test.go

```diff
@@ -142,10 +142,24 @@ func TestHandler(t *testing.T) {
 		})
 
 		t.Run("description=init a flow as browser", func(t *testing.T) {
-			t.Run("description=without privileges", func(t *testing.T) {
-				res, body := initSPAFlow(t, new(http.Client))
-				assert.Equal(t, http.StatusUnauthorized, res.StatusCode, "%s", body)
-				assert.Equal(t, text.ErrNoActiveSession, gjson.GetBytes(body, "error.id").String(), "%s", body)
+			t.Run("case=unauthorized users are redirected to login", func(t *testing.T) {
+				c := testhelpers.NewClientWithCookies(t)
+				// prevent the redirect
+				c.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+					return http.ErrUseLastResponse
+				}
+				returnTo := "?return_to=validRedirect"
+				req, err := http.NewRequest("GET", publicTS.URL+settings.RouteInitBrowserFlow+returnTo, nil)
+				require.NoError(t, err)
+
+				res, err := c.Do(req)
+				require.NoError(t, err)
+				// here we check that the redirect status is 303
+				require.Equal(t, http.StatusSeeOther, res.StatusCode)
+				location, err := res.Location()
+				require.NoError(t, err)
+				require.Equal(t, publicTS.URL+login.RouteInitBrowserFlow+returnTo, location.String())
+				defer res.Body.Close()
```

Comment:
If the `require`s prior to this fail, this is never executed, and thus the `Body` is never closed. 

I don't think we even do this a lot in the other tests, as they're quite short-lived processes and these resource leaks shouldn't really pile up. Nevertheless, it doesn't hurt to do this properly, but it should be moved to right after the `require.NoError` check of the response. See also https://pkg.go.dev/net/http#Client.Do ("On error, any Response can be ignored.").

Reply:
Fixed in https://github.com/ory/kratos/pull/2787/commits/cd252af32c90d6f979286e9ac2bc1d4f1cecd71e